server/api/attendance/attendance.controller.js

```diff
@@ -0,0 +1,674 @@
+// Updates an existing attendance submission in the DB.
+// DON'T USE : does not generate required data for submission
+// Restricted to admins
+// router.put('/:id', auth.hasRole('admin'), controller.update);
+exports.update = function(req, res) {
+    if(req.body._id) { delete req.body._id; }
+    Attendance.findById(req.params.id, function (err, attendance) {
+        if (err) { return handleError(res, err); }
+        if(!attendance) { return res.send(404); }
+        var updated = _.merge(attendance, req.body);
+        updated.save(function (err) {
+            if (err) { return handleError(res, err); }
+            return res.json(200, attendance);
+        });
+    });
+};
+// *******************************************************
+
+// *******************************************************
+// Verifies an existing attendance submission in the DB.
+// Restricted to mentors
+// router.put('/:id/verify', auth.hasRole('mentor'), controller.verifyAttendanceById);
+exports.verifyAttendanceById = function(req, res) {
+    if(req.body._id) { delete req.body._id; }
+    Attendance.findById(req.params.id, function (err, attendance) {
+        if (err) { return handleError(res, err); }
+        if(!attendance) { return res.send(404); }
+        attendance.verified = true;
+        attendance.save(function (err) {
+            if (err) { return handleError(res, err); }
+            return res.json(200, attendance);
+        });
+    });
+};
+// *******************************************************
+
+
+// *******************************************************
+// Verify users large group attendance given submission Id, available only to mentors
+// router.post('/present/:user/full', hasRole('mentor'),controller.verifyFullAttendance);
+exports.verifyFullAttendance = function(req,res){
+    var userId = req.params.id;
+    return ClassYear.getCurrent(function(err, classYear){
+        var classYearId = classYear._id;
+        return getPresent(user._id, isoDateToTime(new Date()), classYear._id, function(userAttendance){
+            if (!userAttendance.length){
+                return res.send(400, 'No attendance found');
+            }
+            for (var a = 0; a < userAttendance.length; a++){
+                if (!userAttendance.smallgroup){
+                    userAttendance[a].verified = true;
+                    userAttendance[a].save();
+                }
+            }
+            return res.send(200);
+
+        });
+    });
+};
+// *******************************************************
+
+// *******************************************************
+// Verify users small group attendance given submission Id, available only to mentors
+// TODO: verify that mentor is part of the smallgroup
+// router.post('/present/:user/small', auth.hasRole('mentor'), controller.verifySmallAttendance);
+exports.verifySmallAttendance = function(req,res){
+    var userId = req.params.id;
+    return ClassYear.getCurrent(function(err, classYear){
```

Comment:
It ensures correct exiting from the function and no write after ends or multiple res.sends.  It is much easier to verify that everything is correct and is one control thread (since there was no promises at the time of writing)